### PR TITLE
Lessons

### PR DESCRIPTION
**Pull request by winter-yuki**
- provide basic UI for all lesson types
- implemented logic of lessons
- some refactoring is done

### CODE REVIEW

**Comment by winter-yuki** [comment]
@zuevval I've done with info fragment logic. Maybe we can say that the current concept with different fragments for lessons works. Check it, please. Maybe you can start working on lesson layouts.

**Comment by winter-yuki** [comment]
Done!

File: android/app/src/main/java/ru/spbstu/amd/learnbraille/screens/lessons/InputSymbolFragment.kt [hunk 1]
```diff
@@ -74,12 +74,6 @@ class InputSymbolFragment : AbstractLesson(R.string.lessons_help_input_symbol) {
             navigateToPrevStep(database.stepDao, step)
         }
 
-        toCurrStepButton.setOnClickListener {
```
Comment (winter-yuki) [inline android/app/src/main/java/ru/spbstu/amd/learnbraille/screens/lessons/InputSymbolFragment.kt]: Why is it removed?

File: android/app/src/main/java/ru/spbstu/amd/learnbraille/screens/lessons/InputSymbolFragment.kt [hunk 2]
```diff
@@ -44,8 +44,8 @@ class InputSymbolFragment : AbstractLesson(R.string.lessons_help_input_symbol) {
 
         val step = getStepArg()
         require(step.data is InputSymbol)
-        titleTextView.text = step.title
-        infoTextView.text = step.data.symbol.symbol.toString()
+        //titleTextView.text = step.title
```
Comment (winter-yuki) [inline android/app/src/main/java/ru/spbstu/amd/learnbraille/screens/lessons/InputSymbolFragment.kt:47]: Why the title is removed?

File: android/app/src/main/java/ru/spbstu/amd/learnbraille/screens/lessons/InputDotsFragment.kt [hunk 3]
```diff
@@ -44,10 +44,8 @@ class InputDotsFragment : AbstractLesson(R.string.lessons_help_input_dots) {
 
         val step = getStepArg()
         require(step.data is InputDots)
-        titleTextView.text = step.title
-        infoTextView.text = step.data.text
-            ?: getString(R.string.lessons_input_dots_info_template)
-                .format(step.data.dots.spelling)
+        //titleTextView.text = step.title
+        letter.text = " "
```
Comment (winter-yuki) [inline android/app/src/main/java/ru/spbstu/amd/learnbraille/screens/lessons/InputDotsFragment.kt:48]: We do not have a letter here. We have some message like "All dots" or generated one "Dots: 1 3 5"

File: android/app/src/main/res/layout/fragment_lessons_input.xml [hunk 4]
```diff
@@ -0,0 +1,233 @@
...
+                    android:layout_weight="1"
+                    android:visibility="invisible" />
+            </LinearLayout>
+
+        </androidx.constraintlayout.widget.ConstraintLayout>
+
+
+    </LinearLayout>
+
+
```
Comment (winter-yuki) [inline android/app/src/main/res/layout/fragment_lessons_input.xml:154]: Please, add TODO here if we not have decided yet,

File: android/app/src/main/res/layout/lessons_buttonbar_left.xml [hunk 5]
```diff
@@ -10,19 +10,6 @@
         android:layout_weight="1"
         android:visibility="invisible" />
 
-    <!---<Button
```
Comment (winter-yuki) [inline android/app/src/main/res/layout/lessons_buttonbar_left.xml]: Add TODO here also

File: android/app/src/main/java/ru/spbstu/amd/learnbraille/screens/lessons/InputDotsFragment.kt (hunk 3 above)
Comment (zuevval) [reply android/app/src/main/java/ru/spbstu/amd/learnbraille/screens/lessons/InputDotsFragment.kt:48]: It should be printed in TitleTextView in smaller font

**Comment by winter-yuki** [comment]
So I think that we should revert last commits until `Fix merge errors` and merge this PR. It is tremendous already.

**Comment by winter-yuki** [comment]
And also I don't want to make decisions on the fly.

**Pull request merged**